docs/sdg/sdg-refactor.md

```diff
@@ -0,0 +1,136 @@
+    - 1.3. They have a taxonomy with some new seed data they added to it.  They want to run SDG only.
+        - 1.3.1. They also want to see the inputs to SDG that get extracted from the taxonomy (i.e., a set of seed context/question/answer tuples and optionallly a set of document chunks).
+        - 1.3.2. Alternatively, maybe they only want to see the inputs to SDG -- they don't actually want to run SDG.
+    - 1.4. They have a set of seed context/question/answer tuples and optionallly a set of document chunks.  They want to run the full pipeline including SDG and model training and evaluation.
+    - 1.5. They have a set of seed context/question/answer tuples and optionallly a set of document chunks.  They want to run SDG and then evaluate an existing model on the outputs of that SDG.
+    - 1.6. They have a set of seed context/question/answer tuples and optionallly a set of document chunks.  They want to run SDG only.
+2. User installs the SDG library only.  They want to invoke any of the following as a library call from code they write:
+    - 2.1. They have a taxonomy with some new seed data they added to it.  They want to run SDG only without any postprocessing.
+    - 2.2. They have a set of seed context/question/answer tuples and optionallly a set of document chunks.  They want to run SDG only without any postprocessing.
+
+If I understand the guidance from our product management, the flows that our users want us to support here are 1.1., 1.2., 1.3, 1.3.1, and 2.2.  However, I am not sure that I understand the guidance.  Are we confident that we do want to support those flows and do not want to support any of the others?  More specifically, do we think the users who want 2.2 might be satisfied with 1.6 instead?  Both of those are SDG-only flows, but the latter is more developer focused and the former is more business-user focused.  Also, are we really confident that all of the "SDG only" customers really want to do their own document chunking?  Might there be some customers that have seed context/question/answer tuples and documents and want us to chunk the documents for them?
+
+Also note, that for all of the flows above that start with "they have a taxonomy", there are open questions around the complexity of the taxonomy format.  Some users want to be able to provide seed data and documents without explicitly or implicitly extending some sort of base taxonomy.  If we had something simpler than the existing taxonomy format that still included some way to specify context/question/answer tuples and references to full documents (i.e., not *chunks*), would we even call that a taxonomy or would it be something else?  If we call it something else, then some or all of the flows that take in a taxonomy might also be applicable for that something else.  For the remainder of this document, however, we will assume that any sort of simplified/easier-to-use variant of a taxonomy will also be called a "taxonomy".
```

Comment:
1. From lab flow, a "taxonomy" is a git repo; as such, it can be cloned/pushed/shared using the normal git constructs and flow. 
2. Another potential flow is the user bringing their documents in an InstructLab-supported schema (in which case, there might not be a traditional taxonomy tree)

It might be good to phrase the proposal such that both flows are possible

Reply:
As a general comment, it seems unlikely that a taxonomy will always be identified with a fit repo. We should distinguish functionality/purpose from mechanism/implementation.